be/src/vec/columns/column_vector.h

```diff
@@ -120,10 +120,13 @@ class ColumnVector final : public COWHelper<ColumnVectorHelper, ColumnVector<T>>
     ColumnVector(std::initializer_list<T> il) : data {il} {}
 
     void insert_res_column(const uint16_t* sel, size_t sel_size, vectorized::ColumnVector<T>* res_ptr) {
+        auto& res_data = res_ptr->data; 
+        res_data.reserve(res_ptr->size() + sel_size);
```

Comment:
why ```+ sel_size```  here?

Reply:
because i will insert sel_size elements into res_ptr's data
the operation of reserve will supply enough memory for assigning one by one in the loop.
each insert_data() call will check whether need to alloc bigger memory chunk in push_back(), that's not highly efficient.

Reply:
```suggestion
        res_data.reserve(sel_size);
```